### ISSUE TITLE
auth session facebook bug

### ISSUE THREAD

**Issue by Stevemoretz** [expo/expo#19208]
### Summary

I've been struggling for the last few hours with facebook authentication (expo go), I have been following this tutorial line by line : 
https://docs.expo.dev/guides/authentication/#facebook

But getting OAuth not whitelisted error again and again, then it occurred to me to log the request and in it I saw it in it was using another project: @community/native-component-list

So I had to change my code to:

```ts
    const [FBrequest, FBresponse, FBpromptAsync] = Facebook.useAuthRequest(
        {
            clientId: "my_client_id",
            responseType: ResponseType.Code,
        },
        {
            projectNameForProxy: "@me/my-project-name",
        }
    );
```

to make it work, the issue is at: node_modules/expo-auth-session/src/providers/Facebook.ts

```ts
    return makeRedirectUri({
      // The redirect URI should be created using fb + client ID on native.
      native: `fb${clientId}://authorize`,
      useProxy,
      projectNameForProxy: '@community/native-component-list',
      ...redirectUriOptions,
    });
```

Here it is : https://github.com/expo/expo/blob/main/packages/expo-auth-session/src/providers/Facebook.ts

I did try reinstalling but no use.

### What platform(s) does this occur on?

iOS

### Environment

```

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 10.15.7
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v16.16.0/bin/yarn
      npm: 8.15.1 - ~/.nvm/versions/node/v16.16.0/bin/npm
      Watchman: 2022.02.14.00 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
      Android SDK:
        API Levels: 28, 29, 30, 32
        Build Tools: 28.0.3, 29.0.2, 29.0.3
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6626763
      Xcode: 12.4/12D4e - /usr/bin/xcodebuild
    npmPackages:
      expo: ^46.0.10 => 46.0.10 
      react: 18.0.0 => 18.0.0 
      react-dom: 18.0.0 => 18.0.0 
      react-native: 0.69.5 => 0.69.5 
      react-native-web: ~0.18.7 => 0.18.9 
    npmGlobalPackages:
      eas-cli: 1.1.1
      expo-cli: 6.0.5
    Expo Workflow: managed

```

### Minimal reproducible example

I can't share my full code because it contains credentials, this is same as my code despite credentials: https://docs.expo.dev/guides/authentication/#facebook

**Comment by pennv** [comment]
same here, happened after upgrading from 3.6.1 to 3.7.1.

just downgraded back and it looks fine for now.

**Comment by Stevemoretz** [comment]
> same here, happened after upgrading from 3.6.1 to 3.7.1.
> 
> just downgraded back and it looks fine for now.

You don't need to downgrade just add your projectNameForProxy like I did. (replace the value with yours)

```ts
 const [FBrequest, FBresponse, FBpromptAsync] = Facebook.useAuthRequest(
        {
            clientId: "my_client_id",
            responseType: ResponseType.Code,
        },
        {
            projectNameForProxy: "@me/my-project-name",
        }
    )
```

**Comment by pennv** [comment]
sorry didn't read through - just added projectNameForProxy and it works like a charm, thank you very much :)

**Comment by Stevemoretz** [comment]
> sorry didn't read through - just added projectNameForProxy and it works like a charm, thank you very much :)

You are welcome :)

**Comment by brentvatne** [comment]
fixed in expo-auth-session@3.7.3 and published as latest for sdk 47. thank you for catching my embarrassing mistake

**Comment by bcarradini** [comment]
It looks like this issue was fixed in expo-auth-session@3.7.3 but re-introduced in expo-auth-session@3.7.4.

When I install 3.7.3, I see this in `node_modules/expo-auth-session/build/providers/Facebook.ts`:
```
    return makeRedirectUri({
      // The redirect URI should be created using fb + client ID on native.
      native: `fb${clientId}://authorize`,
      useProxy,
      ...redirectUriOptions,
    });
```

When I install 3.7.4, I see this in `node_modules/expo-auth-session/build/providers/Facebook.ts`:
```
    return makeRedirectUri({
      // The redirect URI should be created using fb + client ID on native.
      native: `fb${clientId}://authorize`,
      useProxy,
      projectNameForProxy: '@community/native-component-list',
      ...redirectUriOptions,
    });
```
